src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertyAndValueDeserializer.cs

```diff
@@ -327,6 +322,109 @@ internal object ReadODataOrCustomInstanceAnnotationValue(string annotationName,
+        /// 4). <see cref="ODataResourceValue"/>
+        /// 5). <see cref="ODataCollectionValue"/>
+        /// </returns>
+        /// <remarks>
+        /// Pre-Condition:  JsonNodeType.PrimitiveValue   - the value of the property is a primitive value,
+        ///                 JsonNodeType.StartObject      - the value of the property is an object,
+        ///                 JsonNodeType.StartArray       - the value of the property is an array - method will fail in this case.
+        /// Post-Condition: almost anything - the node after the property value.
+        /// </remarks>
+        internal async Task<object> ReadNonEntityValueAsync(
+            string payloadTypeName,
+            IEdmTypeReference expectedValueTypeReference,
+            PropertyAndAnnotationCollector propertyAndAnnotationCollector,
+            CollectionWithoutExpectedTypeValidator collectionValidator,
+            bool validateNullValue,
+            bool isTopLevelPropertyValue,
+            bool insideResourceValue,
+            string propertyName,
+            bool? isDynamicProperty = null)
+        {
+            this.AssertRecursionDepthIsZero();
+            object nonEntityValue = await this.ReadNonEntityValueImplementationAsync(
+                payloadTypeName,
+                expectedValueTypeReference,
+                propertyAndAnnotationCollector,
+                collectionValidator,
+                validateNullValue,
+                isTopLevelPropertyValue,
+                insideResourceValue,
+                propertyName,
+                isDynamicProperty).ConfigureAwait(false);
+            this.AssertRecursionDepthIsZero();
+
+            return nonEntityValue;
+        }
+
+        /// <summary>
+        /// Asynchronously reads the value of the instance annotation.
+        /// </summary>
+        /// <param name="propertyAndAnnotationCollector">The duplicate property names checker instance.</param>
+        /// <param name="name">The name of the instance annotation.</param>
+        /// <returns>
+        /// A task that represents the asynchronous read operation.
+        /// The value of the TResult parameter contains the value of the instance annotation.
+        /// </returns>
+        internal Task<object> ReadCustomInstanceAnnotationValueAsync(PropertyAndAnnotationCollector propertyAndAnnotationCollector, string name)
+        {
+            Debug.Assert(propertyAndAnnotationCollector != null, "propertyAndAnnotationCollector != null");
+            Debug.Assert(!string.IsNullOrEmpty(name), "!string.IsNullOrEmpty(name)");
+
+            object propertyAnnotation;
+            string odataType = null;
+            if (propertyAndAnnotationCollector.GetODataPropertyAnnotations(name)
+                .TryGetValue(ODataAnnotationNames.ODataType, out propertyAnnotation))
+            {
+                odataType = ReaderUtils.AddEdmPrefixOfTypeName(ReaderUtils.RemovePrefixOfTypeName((string)propertyAnnotation));

```

Comment:
Seems like an opportunity to revisit in the future to cut down string allocations.